state/core_access.go

```diff
@@ -81,19 +94,29 @@ func NewCoreAccessor(
 	coreIP,
 	rpcPort string,
 	grpcPort string,
-) *CoreAccessor {
+	granter string,
```

Comment:
Nit but I think this would've made more sense as an ...Option bc we'll end up breaking constructor more and more the more configurability we add, so we may as well do it now to add opts?

Reply:
added options